### DISCUSSION TITLE
[FEATURE] Custom Reloader class

### DISCUSSION THREAD

**Discussion by Kludex** [Kludex/uvicorn#1473] [Ideas]
### Checklist

- [X] There are no similar issues or pull requests for this yet.
- [ ] I discussed this idea on the [community chat](https://gitter.im/encode/community) and feedback is positive.

### Is your feature related to a problem? Please describe.

I want to have my browser reloaded when my code source is updated. 

https://user-images.githubusercontent.com/7353520/145715004-95eea261-7c3e-48c2-af4c-095fbed2154f.mov

 

### Describe the solution you would like.

Ideally, I'd like to have uvicorn to have an option natively, so I could target an url, something like `--reload-url`.

That flag would launch a test browser with Selenium, either using Firefox or Chrome driver (or any other) and would then restart when the watched files are updated.

### Describe alternatives you considered

The ideal solution involves adding some optional dependencies, and I don't think any member wants that, even if it's comfortable. :sweat_smile: 

As an alternative, I think we should be able to choose the reloader class, analogous to how we do for the HTTP, WebSocket, lifespan and loop classes/options. 

### Additional context

I haven't discussed this on the community chat. I just marked because the template makes it mandatory, which I think is correct... If we could make it optional for members it would be cool :sweat_smile: -- EDIT: I could disable the mark after writing the issue :smile:

**Comment by zaheeruddinfaizdl** [comment]
Hi, is there any update on this feature request?

**Reply by Kludex** [reply]
No.
